project/BuildSettings.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2018-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+import sbt._
+import Keys._
+
+// Scalafmt plugin
+import com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin.autoImport._
+
+// SBT Assembly
+import sbtassembly._
+import sbtassembly.AssemblyKeys._
+
+object BuildSettings {
+
+  lazy val compilerOptions = Seq(
+    "-deprecation",
+    "-encoding", "UTF-8",
+    "-feature",
+    "-language:higherKinds",
+    "-unchecked",
+    "-Ypartial-unification",
+    "-Yno-adapted-args",
+    "-Ywarn-dead-code",
+    "-Ywarn-numeric-widen",
+    "-Xfuture",
+    "-Xlint"
+  )
+
+  lazy val javaCompilerOptions = Seq(
+    "-source", "1.8",
+    "-target", "1.8"
+  )
+
+  lazy val formatting = Seq(
+    scalafmtOnCompile := true,
+    scalafmtTestOnCompile := true
+  )
+
+  lazy val assemblyOptions = Seq(
+    target in assembly := file("target/scala-2.11/assembled_jars/"),
```

Comment:
Just a heads-up that since 1.1 SBT has [unified slash syntax](https://developer.lightbend.com/blog/2017-11-30-sbt-1-1-0-RC1-sbt-server/#unified-slash-syntax-of-setting-keys) for scoping. `in` is not yet deprecated, but I believe it will be.